packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -98,19 +98,44 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   /// See also:
   ///  * [SelectionEventType], which contains all of the possible types.
   SelectionResult dispatchSelectionEvent(SelectionEvent event);
+
+  /// Gets the TextSelection from the selected content
+  ///
+  /// Return `null` if nothing is selected, or is not supported by the selectable
+  TextSelection? get textSelection;
+
+  /// Gets the content length of the selectable
+  ///
+  /// Return `null` if the content length is not supported by the selectable
+  int? get contentLength;
 }
 
 /// The selected content in a [Selectable] or [SelectionHandler].
 // TODO(chunhtai): Add more support for rich content.
 // https://github.com/flutter/flutter/issues/104206.
 class SelectedContent {
   /// Creates a selected content object.
-  ///
-  /// Only supports plain text.
-  const SelectedContent({required this.plainText});
+  /// not all selected content supports TextSelection
+  const SelectedContent({required this.plainText, this.textSelection});
 
   /// The selected content in plain text format.
   final String plainText;
+
+  /// The selected content in [TextSelection] format.
+  final TextSelection? textSelection;
```

Comment:
It looks like we aren't initializing any of our `SelectedContent` usages in the framework with a `TextSelection`, is this correct? It looks like we only set this `textSelection` through `copyWith` in `_updateSelectedContentIfNeeded`.

Reply:
I think we should be initializing our usages of `SelectedContent` across the framework with a `TextSelection` when its available to us since if a `Selectable` has a non-null `SelectedContent` this means it has a valid selection.